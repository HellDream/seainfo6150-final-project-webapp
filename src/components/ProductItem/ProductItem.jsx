import React from 'react';
import styles from './ProductItem.module.css';
import { Link } from 'react-router-dom';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';

const ProductItem = (props) => {
    dayjs.extend(relativeTime);
    return (
        <li className={styles.product}>
            <Link
                className={styles.link}
                to={{
                    pathname: `${props.category}/product/${props.product.slug}`,
                    product: props.product,
                }}
            >
                <div className={styles.imageBlock}>
                    <img
                        className={styles.image}
                        src={props.product.imageUrl}
                        alt={`${props.title}`}
                    />
                </div>
                <div className={styles.productInfoBlock}>
                    <h3 className={styles.title}>{props.product.title}</h3>
                    <h4 className={styles.price}>
                        ${props.product.price.toFixed(2)}
                    </h4>
                </div>
                <time
                    className={styles.time}
                    dateTime={props.product.postedTime}
                >
                    {dayjs(props.product.postedTime).fromNow()}
                </time>
            </Link>
        </li>
    );
};

export default ProductItem;
