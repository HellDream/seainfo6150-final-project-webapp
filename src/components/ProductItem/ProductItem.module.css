.product {
    display: flex;
    flex-direction: column;
    border: 1px solid rgb(238, 231, 231);
    border-radius: 10px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.3);
    font-family: Arial, Helvetica, sans-serif;
    justify-content: space-between;
    width: 25rem;
}

.imageBlock {
    height: 14rem;
}

.image {
    border-radius: 10px 10px 0px 0px;
    width: 100%;
    height: 100%;
}

.productInfoBlock {
    padding-top: 0.5rem;
    padding-left: 1rem;
    padding-right: 1rem;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.title {
    margin:0;
    font-size: 24px;
    font-weight: 500;
    text-decoration: none;
    color: black;
}

.price {
    margin:0;
    font-size: 24px;
    font-weight: 500;
    text-decoration: none;
    color: red;
}
.time {
    font-size:15px;
    padding-bottom : 1rem;
    padding-left: 1rem
}