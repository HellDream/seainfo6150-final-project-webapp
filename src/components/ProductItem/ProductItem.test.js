import React from 'react';
import ProductItem from './ProductItem';
import { render } from '@testing-library/react';

import img from '../../images/books.jpg';
import { BrowserRouter } from 'react-router-dom';
const product = {
    slug: 1,
    imageUrl: img,
    title: 'book',
    postedTime: new Date('05 October 2011 14:48 UTC').toISOString(),
    price: 2.33,
};
describe('ProductItem tests', () => {
    it('renders correctly', () => {
        const { container } = render(
            <BrowserRouter>
                <ProductItem category="books" product={product} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
