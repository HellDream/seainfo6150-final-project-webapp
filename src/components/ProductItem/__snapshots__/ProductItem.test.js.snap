// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem tests renders correctly 1`] = `
<div>
  <li
    class="product"
  >
    <a
      class="link"
      href="/books/product/1"
    >
      <div
        class="imageBlock"
      >
        <img
          alt="book"
          class="image"
          src="books.jpg"
        />
      </div>
      <div
        class="productInfoBlock"
      >
        <h3
          class="title"
        >
          book
        </h3>
        <h4
          class="price"
        >
          $
          2.33
        </h4>
      </div>
      <time
        class="time"
        datetime="2011-10-05T14:48:00.000Z"
      >
        9 years ago
      </time>
    </a>
  </li>
</div>
`;
